internal/resource/manage/timeout/timeout_test.go

```diff
@@ -0,0 +1,73 @@
+package timeout_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/slok/kahoy/internal/model"
+	"github.com/slok/kahoy/internal/resource/manage/managemock"
+	"github.com/slok/kahoy/internal/resource/manage/timeout"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTimeoutManagerApply(t *testing.T) {
+	tests := map[string]struct {
+		config    timeout.TimeoutManagerConfig
+		resources []model.Resource
+		mock      func(mrm *managemock.ResourceManager)
+		expErr    bool
+	}{
+		"Basic initialization should not fail.": {
+			mock: func(mrm *managemock.ResourceManager) {
+				mrm.On("Apply", mock.Anything, mock.Anything).Once().Return(nil)
+			},
+		},
+
+		"If Apply takes longer than timeout, apply should fail.": {
+			config: timeout.TimeoutManagerConfig{
+				Timeout: 1 * time.Second,
+			},
+			resources: []model.Resource{
+				{ID: "resource1", GroupID: "group1"},
+			},
+			mock: func(mrm *managemock.ResourceManager) {
+				expBatch := []model.Resource{
+					{ID: "resource1", GroupID: "group1"},
+				}
+				mrm.On("Apply", mock.Anything, expBatch).Once().Return(nil)
+				// TODO(jesus.vazquez) wait few seconds so manager timeouts
```

Comment:
@slok any ideas about how to change the mock so that it waits few seconds and the timeout goes out?

Reply:
I resolved this issue with a custom manager implementation that handles context cancelation.